src/proposals/20241114/SparkBase_20241114.t.sol

```diff
@@ -0,0 +1,124 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.10;
+
+import 'src/SparkTestBase.sol';
+
+import { CCTPForwarder } from "xchain-helpers/forwarders/CCTPForwarder.sol";
+
+import { IALMProxy }         from "spark-alm-controller/src/interfaces/IALMProxy.sol";
+import { IRateLimits }       from "spark-alm-controller/src/interfaces/IRateLimits.sol";
+import { ForeignController } from "spark-alm-controller/src/ForeignController.sol";
+import { RateLimitHelpers }  from "spark-alm-controller/src/RateLimitHelpers.sol";
+
+import { IPSM3 } from "spark-psm/src/interfaces/IPSM3.sol";
+
+contract SparkBase_20241114Test is SparkBaseTestBase {
+
+    address internal constant FREEZER = 0x90D8c80C028B4C09C0d8dcAab9bbB057F0513431;  // Gov. facilitator multisig
+    address internal constant RELAYER = 0x8a25A24EDE9482C4Fc0738F99611BE58F1c839AB;
+
+    constructor() {
+        id = '20241114';
+    }
+
+    function setUp() public {
+        vm.createSelectFork(getChain('base').rpcUrl, 22015320);  // Nov 5, 2024
+        payload = deployPayload();
+    }
+
+    function testALMControllerDeployment() public {
+        // Copied from the init library, but no harm checking this here
+        IALMProxy almProxy           = IALMProxy(Base.ALM_PROXY);
+        IRateLimits rateLimits       = IRateLimits(Base.ALM_RATE_LIMITS);
+        ForeignController controller = ForeignController(Base.ALM_CONTROLLER);
+
+        assertEq(almProxy.hasRole(0x0, Base.SPARK_EXECUTOR), true,   "incorrect-admin-almProxy");
+        assertEq(rateLimits.hasRole(0x0, Base.SPARK_EXECUTOR), true, "incorrect-admin-rateLimits");
+        assertEq(controller.hasRole(0x0, Base.SPARK_EXECUTOR), true, "incorrect-admin-controller");
+
+        assertEq(address(controller.proxy()),      Base.ALM_PROXY,            "incorrect-almProxy");
```

Comment:
Lets add this:


```solidity
        assertEq(almProxy.hasRole(0x0, DEPLOYER), false,   "incorrect-admin-almProxy");
        assertEq(rateLimits.hasRole(0x0, DEPLOYER), false, "incorrect-admin-rateLimits");
        assertEq(controller.hasRole(0x0, DEPLOYER), false, "incorrect-admin-controller");
```

with `address internal constant DEPLOYER = 0x6F3066538A648b9CFad0679DF0a7e40882A23AA4;` at the top

Reply:
Done, but it's important to check the contract itself that no additional admin has been added. These checks are not exhaustive.

Reply:
manually checked that [Executor contract](https://basescan.org/address/0xf93b7122450a50af3e5a76e1d546e95ac1d0f579) is the only role granted access (by events) in:
- [ALM_PROXY](https://basescan.org/address/0x2917956eFF0B5eaF030abDB4EF4296DF775009cA#events)
- [ALM_RATE_LIMITS](https://basescan.org/address/0x983eC82E45C61a42FDDA7B3c43B8C767004c8A74#events)
- [ALM_CONTROLLER](https://basescan.org/address/0xc07f705D0C0e9F8C79C5fbb748aC1246BBCC37Ba#events)